tools/nl/embeddings/finetuning/finetune.py

```diff
@@ -242,17 +283,24 @@ def fine_tune_model(model: Any, df_svs: pd.DataFrame,
 def main(_):
 
   assert FLAGS.model_name_v2 and FLAGS.bucket_name_v2
-  assert FLAGS.embeddings_size in ['small', 'medium']
+  assert (FLAGS.stage == "alternatives") or ((FLAGS.stage == "final") and
+                                             (FLAGS.pretuned_model))
```

Comment:
Two comments:

1. The "final" stage builds off a finetuned model but it should be the finetuned version with only the "alternative" texts (since that takes a long while and we don't expect that to change much right now). All the manually provided pairs should be used as part of the "final" step of fine tuning. The model reference in model.yaml is for the finetuned version which has BOTH steps (alternatives and the manually provided pairs).

2. You suggested in a previous comment on this PR to get rid of the intermediate finetuned model (based on the alternative texts): https://github.com/datacommonsorg/website/pull/2857#discussion_r1240546942 So in order for us to use the most current/checked-in model for the "final" finetuning, we will need this intermediate finetuned model referenced in the model.yaml file. Should I add that back in?

Reply:
Ok resolved this based on our offline chat. Extracting the intermediate model name from the final model name in model.yaml. Also renaming the first stage output to "intermediate".